ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/StatementExecutorTest.java

```diff
@@ -449,6 +451,65 @@ public void shouldEnforceReferentialIntegrity() {
 
   }
 
+  @Test
+  public void shouldFailCreateAsSelectIfExceedActivePersistentQueriesLimit() {
+    // Given:
+    // Create streams and start two persistent queries
+    createStreamsAndTables();
+    // Prepare to try adding a third
+    final KsqlConfig otherConfig =
+        configWith(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG, 2);
```

Comment:
rather than return an instance of `KsqlConfig` and then pass it to other methods, just set a field, (and maybe rename to `givenConfigWith` as it's now not returning state but setting up preconditions. It's probably OK to update `ksqlConfig`. 

If you would rather set a `cmdConfig`, then that would also be fine, but just rename `configWith` to `givenCmdConfig` or similar, so it matches.

Once done, you can remove `otherConfig` from `givenCommand`.